### PR TITLE
Update protoc to v29.1 and protoc-gen-go to v1.36.0

### PR DESCRIPTION
**Pull request by doriable**

### CODE REVIEW

File: make/go/dep_protoc.mk [hunk 1]
```diff
@@ -10,9 +10,9 @@ $(call _assert_var,CACHE_INCLUDE)
 $(call _assert_var,CACHE_BIN)
 
 # Settable
-# https://github.com/protocolbuffers/protobuf/releases 20240523 checked 20240527
+# https://github.com/protocolbuffers/protobuf/releases 20241204 checked 20241217
 # NOTE: Set to version compatible with genproto source code (only used in tests).
-PROTOC_VERSION ?= 27.0
+PROTOC_VERSION ?= 29.1
```
Comment (pkwarren) [inline make/go/dep_protoc.mk:15]: Do we also want to update protoc_gen_go.mk while we're here?

Comment (doriable) [reply make/go/dep_protoc.mk:15]: Yep, this seems reasonable, added `protoc-gen-go` also :)

**Review by pkwarren — APPROVED** [review]

**Review by mfridman — APPROVED** [review]

File: make/go/dep_protoc.mk (hunk 1 above)
Comment (mfridman) [inline make/go/dep_protoc.mk:15]: Did we need to update the script to include any additional files?

https://github.com/bufbuild/makego/blob/e7d3bb3f61509411ebfae10283b677b304f04c72/make/go/dep_protoc.mk/#L47

Comment (doriable) [reply make/go/dep_protoc.mk:15]: It is included as part of `include/google`, I checked by manually testing and upgrade to `bufbuild/buf` and running `make generate`.

Comment (mfridman) [reply make/go/dep_protoc.mk:15]: Yep, checked as well.

```
include/google/protobuf
├── any.proto
├── api.proto
├── compiler
│   └── plugin.proto
├── cpp_features.proto
├── descriptor.proto
├── duration.proto
├── empty.proto
├── field_mask.proto
├── go_features.proto
├── java_features.proto
├── source_context.proto
├── struct.proto
├── timestamp.proto
├── type.proto
└── wrappers.proto

2 directories, 15 files
```

**Pull request merged**